### DISCUSSION TITLE
Internet not working without mitmproxy enabled

### DISCUSSION THREAD

**Discussion by levica99** [mitmproxy/mitmproxy#6741] [Q&A] [answered]
I'm not sure if there's already a fix for this problem, but I could not find anything for the life of me, so I'm resorting to opening an issue. Around 3 days ago I used mitmproxy on my computer and on my phone through adb, just once. I've used mitmproxy a few other times on the PC since then, and I have no issues. On my phone on the other hand, internet access simply stopped working altogether - whether it be Wi-Fi, mobile data or tethering/hotspot. I was trying to fix it, when I noticed it strictly only works if I connect my phone to my computer again, start adb and mitmproxy. How could I fix this? I've already tried many, many things including resetting network settings, setting up a manual IP (and therefore DNS and proxy) for both my Wi-Fi and mobile data connections, nothing worked. I'm pretty sure it would be fixed if I factory reset, but I certainly want to leave it as a very last resort. Help pls.

Edit: I've also tried reconnecting my phone to mitmproxy and disabling it properly of course, restarting my phone, etc. Many of the basic things. It's an android.

**Comment by Prinzhorn** [comment]
Did you use WireGuard mode? Is WireGuard (or any VPN) turned off?

**Reply by levica99** [reply]
I'm not sure what WireGuard mode is, so I'm assuming I didn't use it. And yeah, I have no VPN on nor even installed. I'm considering factory resetting my settings, although it would wipe out all my accounts and personal data in apps; so I'm scared to do that if the problem isn't going to be fixed. Also, there's something that I think could help find out the issue, which is that, for some reason, *some* apps work properly, like Twitch, WhatsApp, and Instagram. Most other apps don't. If I try to access Instagram or Twitch through any browser, I'm not able to connect.

Looking up WireGuard mode, do you think it could help in anyway to try it out? (Just connecting, disconnecting, messing with certificates, I don't know... Although I have no internet connection, so I'm not sure how WireGuard would behave.)

**Reply by Prinzhorn** [reply]
It doesn't make sense to try out WireGuard mode. Your problem is clearly not related to mitmproxy, since it doesn't run on your phone. Either you didn't revert some of the setup on the phone (e.g. remove the system wide proxy or whatever way you chose to connect to mitmproxy) or you ran into an Android bug. I personally don't know how I could be of any more help.

**Reply by levica99** [reply, marked as answer]
That's fair. I was finally able to fix it, it was a problem with the adb (Android Debug Bridge), which is what I used to connect to mitmproxy. Thank you either way, you were of help by leading me into the right possible causes.

For curiosity's sake, it's because I used the shell command `settings put global http_proxy 127.0.0.1:8080` and to remove the proxy, you need to specifically then use `settings put global http_proxy  :0`. For some reason it's the only way to remove a proxy on Android if you set it up through the shell. `settings reset global http_proxy`, `settings delete global http_proxy` or `settings put global http_proxy 0:0` don't work, and they're what I had tried before.
